custom_model_runner/datarobot_drum/drum/language_predictors/base_language_predictor.py

```diff
@@ -127,6 +132,24 @@ def _init_mlops(self):
         if not mlops_loaded:
             raise Exception("MLOps module was not imported: {}".format(mlops_import_error))
 
+        if to_bool(self._params.get("allow_dr_api_access")):
+            try:
+                self._deployment = dr.Deployment.get(deployment_id)
+                drift_settings = self._deployment.get_drift_tracking_settings()
+                self._feature_drift_enabled = drift_settings.get("feature_drift", True).get(
+                    "enabled", True
+                )
+                self._target_drift_enabled = drift_settings.get("target_drift", True).get(
+                    "enabled", True
```

Comment:
I forget, can this stuff be toggled on and off dynamically in a deployment? If so, is this code going to pick up such changes?

Reply:
Good point ! I am checking with #mmm about it .

Reply:
You are right !  The flags can be toggled on / off dynamically and not possible to pick those changes unless you query deployment every time.  But, another way is to check if the training data is assigned or not.  For custom models - assigning training data => new model version => model replacement.  So, we can easily verify during the startup if the training data is assigned to this model or not and then decide to report the predictions.

I will close this PR and open a new one later. I think we also need to handle other aspects of reporting with it.

Reply:
Conversely, you can
- have a background thread that periodically checks the state of the upstream setting
- not needing threads but similar, perform the check on demand but cache the answer from some amount of time (can add some unexpected latency to requests periodically).

Reply:
Either of the solution will be a big architectural change - ideally would like to have bigger motivation than this optimization to drive it.  We do have some other unrelated changes planned and I will see if any of those could be benefitted by a thread or cache.

Reply:
Understood. My thoughts are though that with no training data, it is **always** true that drift tracking will be disabled; however, with training data assigned, it is not always true that drift tracking is enabled (because it can still be turned off at the deployment level for whatever reason) 🤷‍♂️ I'll don't want the perfect to be the enemy of the good though so just wanted to raise these concerns but will defer to you on the ultimate choice.

Reply:
Right ! without training data - drift is definitely off. With training data it can be on or off - but typically will be on.  The worst case where training data is there and drift is off - thats when the API call will be a waste - but that will be a non-typical case.  It will solve more than 80% of the problem.  Currently it always posts - and most of the times for text gen the training data is missing